frontend/src/libs/components/tooltip/element-with-tooltip.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  createElement,
+  type FC,
+  type ReactElement,
+  type ReactHTML,
+  type ReactNode,
+} from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+import { DataTooltipId } from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+
+type ElementWithTooltipProperties = {
+  className?: string;
+  elementType: keyof ReactHTML;
+  placement?: 'bottom' | 'top' | 'left' | 'right';
+  tooltipId?: ValueOf<typeof DataTooltipId>;
+  tooltipContent: string | ReactElement;
+  children?: ReactNode;
+};
+
+const ElementWithTooltip: FC<ElementWithTooltipProperties> = ({
+  children,
+  className,
+  elementType,
+  placement = 'top',
+  tooltipId = DataTooltipId.MAIN_TOOLTIP,
+  tooltipContent,
```

Comment:
Relates to suggestion in achievement-list
```suggestion
  hasToShowTooltip = true,
  tooltipContent,
```

Reply:
[872c299](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/156/commits/872c299d0e15e929c365b18e9248a493ea3455fc)